### PR TITLE
Laura/78 add algolia search support 1

### PR DESCRIPTION
**Pull request by LauraLangdon**
Referred to env vars for DocSearch secrets.

### CODE REVIEW

**Comment by arbourd** [comment]
It seems that the build step is failing due to:
```
Error:  ValidationError: "algolia.apiKey" is required
```

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -71,11 +71,12 @@ module.exports = {
...
-//      apiKey: '',
-//      indexName: '',
-//      contextualSearch:
-//    },
+   algolia: {
+     appId: process.env.DOCSEARCH_APP_ID,
+     apiKey: process.env.DOCSEARCH_API_KEY,
+     indexName: 'suborbital',
+     contextualSearch: true
+     },
```
Comment (arbourd) [inline website/docusaurus.config.js:79]:
```suggestion
    },
```

Comment (arbourd) [inline website/docusaurus.config.js:79]: some spacing stuff
```suggestion
    algolia: {
      appId: process.env.DOCSEARCH_APP_ID,
      apiKey: process.env.DOCSEARCH_API_KEY,
      indexName: 'suborbital',
      contextualSearch: true
    },
```

**Comment by LauraLangdon** [comment]
> It seems that the build step is failing due to:
> 
> ```
> Error:  ValidationError: "algolia.apiKey" is required
> ```
> 
> This is probably because the env var is `null` or `""`, which makes sense because we haven't provided it in this PR.
> 
> #119 would need to be merged first but you can combine them into one PR here which would test the whole flow.

I've searched up "combine pull requests" but I'm mostly finding GitHub actions to do this. Is there a specific way you had in mind?

**Comment by LauraLangdon** [comment]
Woohoo, the connection is made! [Obligatory Elastica](https://youtu.be/ilKcXIFi-Rc) (why doesn't YouTube have an embed option for GitHub? 😛).

I'll work on setting up a crawl on Monday.

**Comment by LauraLangdon** [comment]
The tests all passed on Friday but failed today, so something's buggy. Maybe related: I'm now getting ```[ERROR] ValidationError: "algolia.apiKey" is required``` when I run `npm start` on the branch I've been working on this in, despite having set up the secrets in GitHub for for the API key and the appID. 🤔

**Review by flaki — CHANGES_REQUESTED** [review]
There is that pesky `package-lock.json` again! You will need to remove those changes from the PR.

Also let's move the link checker to its own PR.

Looks good otherwise!

I have updated the docs to use the latest Docusaurus Beta 18 to make sure the crawler on the public site gets everything right. Because of this you will need to rebase the branch on `main` (and will probably see conflicts due to package-lock.json, you should be able to discard your local changes and go with the file on `main`).

I had to tweak the [crawler configuration](https://crawler.algolia.com/admin/crawlers/) from the [original](https://docsearch.algolia.com/docs/templates/#docusaurus-v2-template) to have it correctly pick up the headings, it's still not perfect but much better, and we should spend some thought on #64 and generally on tweaking the docs structure in the coming weeks.

image
[The search in the preview now works as expected](https://suborbital.github.io/docs/laura/78-add-algolia-search-support-1/) so let's get this merged!

**Comment by LauraLangdon** [comment]
> There is that pesky `package-lock.json` again! You will need to remove those changes from the PR.

Oops! Will do.
 
> Also let's move the link checker to its own PR.

The PR for the link checker is already merged; I'm not sure how it ended up here. 
 
> I have updated the docs to use the latest Docusaurus Beta 18 to make sure the crawler on the public site gets everything right. Because of this you will need to rebase the branch on `main` (and will probably see conflicts due to package-lock.json, you should be able to discard your local changes and go with the file on `main`).

Great! I'll give that a go.

> I had to tweak the [crawler configuration](https://crawler.algolia.com/admin/crawlers/) from the [original](https://docsearch.algolia.com/docs/templates/#docusaurus-v2-template) to have it correctly pick up the headings, it's still not perfect but much better, and we should spend some thought on #64 and generally on tweaking the docs structure in the coming weeks.
> 

I'll check that out!

> image [The search in the preview now works as expected](https://suborbital.github.io/docs/laura/78-add-algolia-search-support-1/) so let's get this merged!

Woohoo!

File: .github/workflows/links.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1.1.1
+        with:
+          args: --verbose --no-progress -t 5 --exclude-mail --exclude=local.suborbital.network *.md docs/**/*.md
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Fail if there were link errors
+        run: exit ${{ steps.lychee.outputs.exit_code }}
```
[the remark this answers is not in the document]
Comment (LauraLangdon) [reply .github/workflows/links.yml]: Done!

**Comment by flaki** [comment]
Closing in favor of #126

**Pull request closed without merging**